app/src/main/java/org/mozilla/fenix/tabtray/TabTrayFragment.kt

```diff
@@ -0,0 +1,243 @@
+import mozilla.components.feature.tabs.tabstray.TabsFeature
+import kotlinx.android.synthetic.main.fragment_tab_tray.tabsTray
+import kotlinx.android.synthetic.main.fragment_tab_tray.view.*
+import mozilla.components.support.base.feature.UserInteractionHandler
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.requireComponents
+import androidx.navigation.Navigation.findNavController
+import androidx.navigation.fragment.findNavController
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.tabstray.Tab
+import mozilla.components.concept.tabstray.TabsTray
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.collections.SaveCollectionStep
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.nav
+import org.mozilla.fenix.ext.sessionsOfType
+import org.mozilla.fenix.ext.showToolbar
+
+@SuppressWarnings("TooManyFunctions", "LargeClass")
+class TabTrayFragment : Fragment(), TabsTray.Observer, UserInteractionHandler {
+    private var tabsFeature: TabsFeature? = null
+    var tabTrayMenu: Menu? = null
+
+    private val sessionManager: SessionManager
+        get() = requireComponents.core.sessionManager
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
+        inflater.inflate(R.layout.fragment_tab_tray, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        showToolbar(getString(R.string.tab_tray_title))
+        onTabsChanged()
+
+        sessionManager.register(observer = object : SessionManager.Observer {
+            override fun onSessionAdded(session: Session) {
+                onTabsChanged()
+            }
+
+            override fun onSessionRemoved(session: Session) {
+                onTabsChanged()
+            }
+
+            override fun onSessionsRestored() {
+                onTabsChanged()
+            }
+
+            override fun onAllSessionsRemoved() {
+                onTabsChanged()
+            }
+        }, owner = viewLifecycleOwner)
+
+        tabsFeature = TabsFeature(
+            tabsTray,
+            requireComponents.core.store,
+            requireComponents.useCases.tabsUseCases,
+            { it.content.private == (activity as HomeActivity?)?.browsingModeManager?.mode?.isPrivate },
+            ::closeTabsTray)
+
+        view.tab_tray_open_new_tab.setOnClickListener {
+            val directions = TabTrayFragmentDirections.actionGlobalSearch(null)
+            val navController = findNavController(it)
+            navController.navigate(directions)
```

Comment:
nit - can't we just use `findNavController().navigate` since we're in a fragment for all of these?